pyuvdata/uvbase.py

```diff
@@ -260,11 +274,19 @@ def __eq__(self, other, check_extra=True):
                         f" right is {other_extra}."
                     )
                     return False
-
                 p_check = self_required + self_extra
             else:
                 p_check = self_required
 
+            if allowed_failures is not None:
+                for i, param in enumerate(allowed_failures):
```

Comment:
We never actually do any input verification on `allowed_failures`, should we? It's a lot of work for something so far down in the weeds but if only a string is given (maybe because I only want one to be allowed to fail) then this will iterate through the characters

Reply:
yeah, we probably should make sure it's a list first.

Reply:
Sure, I can turn it into a list if it's a string. Good suggestion.

Reply:
Updated in 00cb301.